app_flutter/lib/status_grid.dart

```diff
@@ -29,27 +45,61 @@ class StatusGrid extends StatelessWidget {
         child: Container(
           width: columnCount * 50.0,
           child: GridView.builder(
-            // TODO(chillers): implement custom scroll physics to match horizontal scroll
-            itemCount: columnCount * commitCount,
+            itemCount: columnCount * statuses.length,
             gridDelegate: SliverGridDelegateWithFixedCrossAxisCount(
                 crossAxisCount: columnCount),
-            itemBuilder: (BuildContext context, int index) {
-              // TODO(chillers): Use StageModel data
-              if (index % columnCount == 0) {
-                return CommitBox(
-                    commit: Commit()
-                      ..author = 'AuthoryMcAuthor #$index'
-                      ..authorAvatarUrl =
-                          'https://avatars2.githubusercontent.com/u/2148558?v=4'
-                      ..repository = 'flutter/cocoon'
-                      ..sha = 'sha shank redemption');
+            itemBuilder: (BuildContext context, int gridIndex) {
+              int statusIndex = gridIndex ~/ columnCount;
+
+              if (gridIndex % columnCount == 0) {
+                return CommitBox(commit: statuses[statusIndex].commit);
               }
 
-              return ResultBox(message: 'Succeeded');
+              return TaskBox(
+                task: _mapGridIndexToTaskBruteForce(gridIndex, columnCount),
+              );
             },
           ),
         ),
       ),
     );
   }
+
+  /// Returns the number of columns the grid should show based on [CommitStatus].
+  ///
+  /// [CommitStatus] is composed of [List<Stage>] that contains the tasks run
+  /// for that stage. Each [Stage] runs a different amount of tasks.
+  ///
+  /// Additionally, [Commit] from [CommitStatus] must have a cell reserved on
+  /// the grid which is another index offset that is accounted.
+  int _getColumnCount(CommitStatus status) {
+    int columnCount = 1; // start at 1 to reserve room for CommitBox column
+
+    for (Stage stage in status.stages) {
+      columnCount += stage.tasks.length;
+    }
+
+    return columnCount;
+  }
+
+  /// Maps a [gridIndex] to a specific [Task] in [List<CommitStatus>]
+  Task _mapGridIndexToTaskBruteForce(int gridIndex, int columnCount) {
```

Comment:
I think this is okay to land as a first draft, but you should pretty quickly concentrate on getting something better running .... unless it doesn't show up in the chrome profile :)

Reply:
I hoped something with "BruteForce" would entice any engineer to improve it :)